katana/native/file_manager/file_manager_utils/backend.py

```diff
@@ -0,0 +1,121 @@
+
+def getpath(request):
+    """
+    This function is to get the path of the User-data location.
+    Whatever is pointed to by this path is displayed.
+    Currently, a static path is read from the file in my VM and the contents are displayed.
+    The file is opened and the path is read
+
+    :return: The path to user-data space
+    """
+    Object = user_utils.UserData('File Manager')
+    print(Object.get_user_path(request))
+    return Object.get_user_path(request)
+
+def delete(path):
+    """
+    This will check if it's a file or directory and then delete them.
+
+    :param path: Full path to File or Directory which has to be deleted.
+    :return: A string saying success or error
+    """
+    if (os.path.isdir(path)):
+        try:
+            shutil.rmtree(path)
+            return 'success'
+        except OSError:
+            print('Cannot delete Directory')
+    elif (os.path.isfile(path)):
+        try:
+            os.remove(path)
+            return 'success'
+        except OSError:
+            print('Cannot delete File')
+
+
+
+def rename(file_path, old_name, new_name):
+    """
+    This function will rename the file or directory.
+
+    :param file_path: The full path of the file or directory
+    :param old_name: Old Name of the file or directory
+    :param new_name: New Name of file or directory
+    :return: String containing success or error.
+    """
+    new_file_path = os.path.join(os.path.dirname(file_path), new_name)
+    try:
+        os.rename(file_path, new_file_path)
+    except os.error as e:
+        print(str(e))
+        return str(e)
+    return 'success'
+
+def save(file_name, username, host, port, destdir, transfer_proto, request):
+    """
+    This function is to save the user-ftp-login details. Password is not stored.
+    A file is created and the details are written into this file.
+
+    :param file_name: File name
+    :param username: User Input
+    :param host: User Input
+    :param port: User Input. Default is 21.
+    :param destdir: User Input. Default is ''
+    :param transfer_proto: User Input. Default is FTP
+    :return: String error or list of cached files in .data directory
+    """
+    parent_path = os.path.join(getpath(request), "File Manager/.data")
+    file_path = os.path.join(parent_path, file_name)
+    try:
+        os.chdir(parent_path)
+    except:
+
+        os.chdir(getpath(request))
+        try:
+            os.mkdir("File Manager")
+            os.chdir(os.path.join(os.getcwd(),"File Manager"))
+        except:
+            os.chdir(os.path.join(os.getcwd(), "File Manager"))
+        os.mkdir(".data")
+        os.chdir(parent_path)
+    try:
+        result = []
+        fp = open(file_name, 'w')
+        write_lines = [username, host, str(port), transfer_proto, destdir]
+        write_lines = '\n'.join(write_lines)
+        fp.writelines(write_lines)
+        fp.close()
+        result = os.listdir(parent_path)
+        return result
+    except:
```

Comment:
Please catch the exception being thrown and either print it out on the console or return it to the view so as to be rendered as an alert on the browser.

Reply:
Done